### PR TITLE
Update docs to new measurement format

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Proposed change(s)

- Update the docs to reflect the new format for measurements
- Remove `constellation config fetch-measurements` from the required steps for creating a new cluster.
  - For new releases the expected measurements of the current release should be directly embedded in the CLI
- Add migration notice for measurements in the config 

### Checklist

- [x] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Link to Milestone

### CODE REVIEW

File: .github/docs/release.md [hunk 1]
```diff
@@ -93,20 +93,23 @@ This checklist will prepare `v1.3.0` from `v1.2.0`. Adjust your version numbers
...
-    13. Create a new tag on this release branch
+    13. Create a new tag on this release branch.
 
         ```sh
         git tag v$ver
         git tags --push
         ```
 
-        * Run [Release CLI](https://github.com/edgelesssys/constellation/actions/workflows/release-cli.yml) action on the tag
+    14. Update expected measurements in [`measurements.go`](/internal/attestation/measurements/measurements.go) using the generated measurements from step 12 and **push your changes**.
```
Comment (datosh) [inline .github/docs/release.md:103]: I think we need to swap 13 & 14, or else the commit to `measurements.go` will not be included in the tag, and therefore the `release-cli.yml` pipeline run won't include it.

Comment (daniel-weisse) [reply .github/docs/release.md:103]: You're right.
Had creating the tag confused with creating the release branch

File: docs/docs/workflows/verify-cluster.md [hunk 2]
```diff
@@ -11,18 +11,57 @@ constellation config fetch-measurements
...
+
 This is because only a subset of the [available measurements](../architecture/attestation.md#runtime-measurements) can be locally reproduced and verified.
 
 During attestation, the validating side (CLI or [join service](../architecture/components.md#joinservice)) compares each measurement reported by the issuing side (first node or joining node) individually.
-For mismatching measurements that are only set under the `measurements` key a warning is emitted.
-For mismatching measurements that are additionally set under `enforcedMeasurements` an error is emitted and attestation fails.
-If attestation fails, the new node can't join the cluster.
+For mismatching measurements that have set `warnOnly` to `true` only warning is emitted.
+For mismatching measurements that have set `warnOnly` to `false` an error is emitted and attestation fails.
+If attestation fails for a newly joining node, the new node isn't permitted to join the cluster.
```
Comment (m1ghtym0) [inline docs/docs/workflows/verify-cluster.md:64]:
```suggestion
If attestation fails for a new node, it isn't permitted to join the cluster.
```

**Review by m1ghtym0 — APPROVED** [review]
Other, than @datosh's comment just a minor nitpick.

**Pull request merged**